clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractFetch.java

```diff
@@ -408,7 +388,21 @@ protected Map<Node, FetchSessionHandler.FetchRequestData> prepareFetchRequests()
         long currentTimeMs = time.milliseconds();
         Map<String, Uuid> topicIds = metadata.topicIds();
 
-        for (TopicPartition partition : fetchablePartitions()) {
+        // This is the set of partitions that have buffered data
+        Set<TopicPartition> buffered = Collections.unmodifiableSet(fetchBuffer.bufferedPartitions());
+
+        // This is the set of partitions that do not have buffered data
+        Set<TopicPartition> unbuffered = Set.copyOf(subscriptions.fetchablePartitions(tp -> !buffered.contains(tp)));
+
+        if (unbuffered.isEmpty()) {
+            // If there are no partitions that don't already have data locally buffered, there's no need to issue
+            // any fetch requests at the present time.
+            return Collections.emptyMap();
```

Comment:
to confirm, after we return from this prepareFetchRequests would be invoked in the next poll()?

Reply:
Correct.